### PR TITLE
{Config} Add endpoint suffix for mysql and postgre sql

### PR DESCRIPTION
**Pull request by handsomesun**
**Description**  
Azure Postgre SQL and MySQL has different endpoint suffix than sql server on public cloud so creating separate configs for them.

Currently we are using the sqlserver_hostname for azure mysql and azure postgresql which is incorrect:
image
(https://msdata.visualstudio.com/Vienna/_git/vienna?path=%2Fsrc%2Fazureml-api%2Fdeploy%2Fspecs%2Fservices-all-base.yaml&version=GBmaster&line=321&lineEnd=325&lineStartColumn=1&lineEndColumn=98&lineStyle=plain)

in public cloud the suffix is azure.com for them not windows.net. which is why i need to create separate config values for them to handle this case. 

Those values are only used by AML services to know the endpoint for mysql/postgresql for connection.

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S169

File: src/azure-cli-core/azure/cli/core/cloud.py [hunk 1]
```diff
@@ -242,6 +248,8 @@ def __str__(self):
         storage_endpoint='core.windows.net',
         keyvault_dns='.vault.azure.net',
         sql_server_hostname='.database.windows.net',
+        mysql_server_hostname='.database.azure.com',
+        postgresql_server_hostname='.database.azure.com',
```
Comment (arrownj) [inline src/azure-cli-core/azure/cli/core/cloud.py:252]: Are they always the same ?

**Comment by Juliehzl** [comment]
1. Please fix title format as report shown.
2. Please fix style error too.
3. To fix the test error, you need to sync with latest dev again.

**Comment by jiasli** [comment]
Transferring from https://github.com/Azure/azure-cli/pull/13076#issuecomment-616911889:

Hi @handsomesun, I have several questions for you change, and hope you can give me more context for this.
1. Currently you only add suffix info in cloud config? Is there any usage in mysql/postgres command for the hostname?

2. You just mentioned that you do the change because mysql/postgres is different form sql hostname. Is there any scenario that we use sql server host name for mysql/postgres?

3. With your change, the only difference I can see is for `az cloud show` command, is this expected?
image

File: src/azure-cli-core/azure/cli/core/cloud.py (hunk 1 above)
Comment (handsomesun) [reply src/azure-cli-core/azure/cli/core/cloud.py:252]: yes for now, but I am not sure if they will change in the future.

**Comment by handsomesun** [comment]
> Transferring from [#13076 (comment)](https://github.com/Azure/azure-cli/pull/13076#issuecomment-616911889):
> 
> Hi @handsomesun, I have several questions for you change, and hope you can give me more context for this.
> 
> 1. Currently you only add suffix info in cloud config? Is there any usage in mysql/postgres command for the hostname?
> 2. You just mentioned that you do the change because mysql/postgres is different form sql hostname. Is there any scenario that we use sql server host name for mysql/postgres?
> 3. With your change, the only difference I can see is for `az cloud show` command, is this expected?
>    image

Currently you only add suffix info in cloud config? Is there any usage in mysql/postgres command for the hostname?
Let me check with Yan for this. 
You just mentioned that you do the change because mysql/postgres is different form sql hostname. Is there any scenario that we use sql server host name for mysql/postgres?
https://msdata.visualstudio.com/Vienna/_git/vienna?path=%2Fsrc%2Fazureml-api%2Fdeploy%2Fspecs%2Fservices-all-base.yaml&version=GBmaster&line=321&lineEnd=325&lineStartColumn=1&lineEndColumn=98&lineStyle=plain

With your change, the only difference I can see is for az cloud show command, is this expected?
Let me check with Yan

**Comment by Juliehzl** [comment]
@shinim to confirm with mysql and postgres module usage.

**Comment by shinilm** [comment]
Another thing  the endpoing suffix we exposed for the posgres is .postgres.database.azure.com and for myql is  .mysql.database.azure.com and for .mariadb.database.azure.com.  We wanted to understand the usage scenario better here

**Comment by Juliehzl** [comment]
@shinilm Could you please take a look again? If no problem, I will merge the PR.

File: src/azure-cli-core/azure/cli/core/cloud.py [hunk 2]
```diff
@@ -242,6 +251,9 @@ def __str__(self):
         storage_endpoint='core.windows.net',
         keyvault_dns='.vault.azure.net',
         sql_server_hostname='.database.windows.net',
+        mysql_server_endpoint='mysql.database.azure.com',
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/cloud.py:254]: No dot `.` prefix?

**Pull request merged**